staging/src/k8s.io/apiserver/pkg/server/healthz/healthz.go

```diff
@@ -253,7 +263,25 @@ func handleRootHealthz(checks ...HealthChecker) http.HandlerFunc {
 		}
 
 		individualCheckOutput.WriteTo(w)
-		fmt.Fprint(w, "healthz check passed\n")
+		fmt.Fprintf(w, "%s check passed\n", name)
+
+		// signal first time this is healthy
+		select {
+		case <-hasBeenHealthy:
```

Comment:
what do we need this block? Is `sync.Once` not thread-safe ?

Reply:
let me just use sync.Once here. I think we don't have to optimize like I tried here.

Reply:
done